azure-ts-aks-keda/keda.ts

```diff
@@ -0,0 +1,170 @@
+
+        const dockercfg = pulumi.all([registry.loginServer, registry.adminUsername, registry.adminPassword])
+            .apply(([server, username, password]) => {
+                const r: any = {};
+                r[server] = {
+                    email: "notneeded@notneeded.com",
+                    username,
+                    password,
+                };
+                return r;
+            });
+
+        // Storage the docker registry credentials as a secret
+        const secretRegistry = new k8s.core.v1.Secret("registry-secret", {
+            metadata: {
+                name: "registry",
+                namespace: "default",
+            },
+            data: {
+                ".dockercfg": dockercfg.apply(c => Buffer.from(JSON.stringify(c)).toString("base64")),
+            },
+            type: "kubernetes.io/dockercfg",
+        },
+        { provider: args.k8sProvider, parent: this });
+
+        // Deploy a KEDA Edge Helm chart
+        const keda = new k8s.helm.v2.Chart("keda-edge", {
+            namespace: "default",
+            repo: "kedacore",
+            chart: "keda-edge",
+            version: "0.0.1-2019.07.24.21.37.42-8ffd9a3",
+            values: {
+                logLevel: "debug",
+            },
+        }, { providers: { kubernetes: args.k8sProvider }, parent: this });
+
+        this.k8sProvider = args.k8sProvider;
+        this.registry = registry;
+        this.registerOutputs();
+    }
+}
+
+// Arguments for an Azure Function App that processes messages from a given storage queue.
+export interface KedaStorageQueueHandlerArgs {
+    resourceGroup: azure.core.ResourceGroup;
+    edge: KedaEdge;
+    storageAccount: azure.storage.Account;
+    queue: azure.storage.Queue;
+    path: pulumi.Input<string>;
+}
+
+// A KEDA-deployed Azure Function App that processes messages from a given storage queue.
+export class KedaStorageQueueHandler extends pulumi.ComponentResource {
+    constructor(name: string,
+                args: KedaStorageQueueHandlerArgs,
+                opts: pulumi.ComponentResourceOptions = {}) {
+        super("examples:keda:KedaStorageQueueHandler", name, args, opts);
+
+        const registry = args.edge.registry;
+
+        // Deploy the docker image of the Function App
+        const dockerImage = new docker.Image("image", {
+            imageName: pulumi.interpolate`${registry.loginServer}/${args.queue.name}:v1.0.0`,
+            build: {
+                context: args.path,
+            },
+            registry: {
+                server: registry.loginServer,
+                username: registry.adminUsername,
+                password: registry.adminPassword,
+            },
+        }, { parent: this });
+
+        // Put the storage account connection string into a secret
+        const secretName = `${name}-connection`;
+        const secretQueue = new k8s.core.v1.Secret("queue-secret", {
+            metadata: {
+                name: secretName,
```

Comment:
Any reason not to use auto-named resources in this class?

Reply:
I copied from a TF example and wasn't sure what the best practice is here. This name should be referenced from `deployment`'s `envFrom`. Should I change that one to `envFrom: [{ secretRef: {name: secretQueue.metadata.name } }]`?

Reply:
Yeah, I would avoid manually specifying the names if possible. Otherwise, it forces downtime if the resources are changed later [1]

[1] https://www.pulumi.com/docs/reference/programming-model/#autonaming